File: poc-kronos-frontend/src/model/curso/portaria/Portaria.js
class Portaria {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataPublicacao = dados ? dados.dataPublicacao : null;
    this.dataInicioVigencia = dados ? dados.dataInicioVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.descricao = dados ? dados.descricao : null;
    this.documentoPublicacao = dados ? dados.documentoPublicacao : null;
    this.idTipoPortaria = dados ? dados.idTipoPortaria : null;
    this.nomeTipoPortaria = dados ? dados.nomeTipoPortaria : null;
    this.idCurso = dados ? dados.idCurso : null;
  }

  static validations(required, maxLength, numeric, dataFimMenor) {
    return {
      documentoPublicacao: { required, maxLength: maxLength(200) },
      descricao: { required, maxLength: maxLength(200) },
      dataPublicacao: { required },
      dataInicioVigencia: { required },
      dataFinalVigencia: { dataFimMenor: dataFimMenor("dataInicioVigencia") },
      idTipoPortaria: { required },
      idCurso: { required }
    };
  }

  static validationMessages() {
    return {
      documentoPublicacao: {
        required: "Documento é obrigatório",
        maxLength: "Documento só aceita no máximo 200 caracteres"
      },
      descricao: {
        required: "Descrição é obrigatória",
        maxLength: "Descrição só aceita no máximo 200 caracteres"
      },
      dataPublicacao: {
        required: "Data Publicação é obrigatório"
      },
      dataInicioVigencia: {
        required: "Data Inicio Vigência é obrigatório"
      },
      dataFinalVigencia: {
        dataFimMenor:
          "Data Final de Vigência não pode ser menor que a Data Início de Vigência"
      },
      idTipoPortaria: {
        required: "Tipo de Portaria é obrigatória"
      },
      idCurso: {
        required: "Curso é obrigatório"
      }
    };
  }
}
export default Portaria;

File: src/main/java/br/com/kronos/backend/aplicacao/curso/Portaria.java
package br.com.kronos.backend.aplicacao.curso;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;
import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirTamanho;

import java.time.LocalDate;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Portaria {
    
    private long id;
    private LocalDate dataPublicacao;
    private LocalDate dataInicioVigencia;
    private LocalDate dataFinalVigencia;   
    private String descricao;
    private String documentoPublicacao;
    private Integer idTipoPortaria;
    private Long idCurso;

	@Builder
	public Portaria(long id, LocalDate dataPublicacao, LocalDate dataInicioVigencia, LocalDate dataFinalVigencia, String descricao, 
                    String documentoPublicacao, Integer idTipoPortaria, Long idCurso ) {

        this.id = id;
        this.dataPublicacao = exigirNaoNulo(dataPublicacao, "Data de publicação");
        this.dataInicioVigencia = exigirNaoNulo(dataInicioVigencia, "Data de início de vigência");
        this.dataFinalVigencia = dataFinalVigencia;
        this.descricao = exigirTamanho(descricao, "Descrição", 1, 200);
        this.documentoPublicacao = exigirTamanho(documentoPublicacao, "Documento de publicação", 1, 200);
        this.idTipoPortaria = exigirNaoNulo(idTipoPortaria, "Tipo de portaria"); 
        this.idCurso = exigirNaoNulo(idCurso, "Curso");      
	}
}